### ISSUE TITLE
"only_once" not respected by "Sleep switch".

### ISSUE THREAD

**Issue by maxi1134** [basnijholt/adaptive-lighting#737]
# Home Assistant Adaptive Lighting Issue Template

## Bug Reports

If you need help with using or configuring Adaptive Lighting, please [open a Q&A discussion thread here](https://github.com/basnijholt/adaptive-lighting/discussions/new?category=q-a) instead.

### Before submitting a bug report, please follow these troubleshooting steps:

Please confirm that you have completed the following steps:

- [ ] I have updated to the [latest Adaptive Lighting version](https://github.com/basnijholt/adaptive-lighting/releases) available in [HACS](https://hacs.xyz/).
- [ ] I have reviewed the [Troubleshooting Section](https://github.com/basnijholt/adaptive-lighting#troubleshooting) in the [README](https://github.com/basnijholt/adaptive-lighting#readme).
- [ ] (If using Zigbee2MQTT) I have read the [Zigbee2MQTT troubleshooting guide](https://github.com/basnijholt/adaptive-lighting#zigbee2mqtt) in the [README](https://github.com/basnijholt/adaptive-lighting#readme).
- [ ] I have checked the [V2 Roadmap](https://github.com/basnijholt/adaptive-lighting/discussions/291) and [open issues](https://github.com/basnijholt/adaptive-lighting/issues) to ensure my issue isn't a duplicate.

### Required information for bug reports:

Please include the following information in your issue.

*Issues missing this information may not be addressed.*

1.  **Debug logs** captured while the issue occurred. [See here for instructions on enabling debug logging](https://github.com/basnijholt/adaptive-lighting#troubleshooting):

```

```

2.  [Your Adaptive Lighting configuration](https://github.com/basnijholt/adaptive-lighting#gear-configuration):

```

```

3.  (If using Zigbee2MQTT), provide your configuration files (**remove all personal information before posting**):
    - `devices.yaml`
    - `groups.yaml`
    - `configuration.yaml` ⚠️; **Warning** _**REMOVE ALL of the PERSONAL INFORMATION BELOW before posting**_ ⚠️;
      - mqtt: `server`:
      - mqtt: `user`:
      - mqtt: `password`:
      - advanced: `pan_id`:
      - advanced: `network_key`:
      - anything in `log_syslog` if you use this
    - Brand and model number of problematic light(s)
```

```

4.  Describe the bug and how to reproduce it:

5. Steps to reproduce the behavior:

**Comment by basnijholt** [comment]
This is actually by design.

Setting these all have the same effect:
- turning light on and off
- turning the adaptive lighting switch on and off
- turning sleep mode switch on and off

It would be much more difficult to implement tracking `only_once` in all cases.

**Comment by maxi1134** [comment]
Would there not be a way for the "to sleep settings" action call on the bulbs check that they are not on?

**Comment by basnijholt** [comment]
Yes, that is possible, however, I would need to implement it in a way that it is configurable because I do not want to change the current behavior (which people rely on).

Currently, I do not find it worth the trouble given the exploding number of configuration options.

What is your use-case if I might ask? 😄

**Comment by maxi1134** [comment]
My use-case is very specific, so probably won't apply to much, but here I go; 

I host events very regularly in my house, events where the lights have saturated colors.

I would like to enable the "Sleep mode" when the main users or the house are asleep ( Partners and Squatters ), while keeping the lights that are still on in their original color until that room is emptied by the remaining guests, turning to a "sleep mode" color when motion is re-detected 
( This will incentivize the guests to remain in the already occupied rooms and slowly make more and more rooms darker. )

My goal will all of this is to automate the programmed death of the party, where people are cordially pushed out by automated means.

**Comment by basnijholt** [comment]
I might be able to implement only_once differently where it marks a light as manually_control. Then I could provide an option where one selects how manually control is reset. However, it might be a while until I find the time for this.

**Comment by maxi1134** [comment]
> Sounds like a cool use case.
> 
> I might be able to implement only_once differently where it marks a light as manually_control. Then I could provide an option where one selects how manually control is reset. However, it might be a while until I find the time for this.

I'm a patient person!
